modules/utils/tests/watchdog_tests.cpp

```diff
@@ -0,0 +1,42 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#include <gtest/gtest.h>
+
+#include "hephaestus/utils/timing/watchdog.h"
+
+// NOLINTNEXTLINE(google-build-using-namespace)
+using namespace ::testing;
+
+namespace heph::utils::timing::tests {
+TEST(Watchdog, TimerFiring) {
+  constexpr auto PERIOD = std::chrono::milliseconds{ 10 };
+  WatchdogTimer timer;
+  std::atomic<int> count{ 0 };
+  timer.start(PERIOD, [&count]() { ++count; });
+  EXPECT_EQ(count, 0);
+  std::this_thread::sleep_for(PERIOD * 4);
+  timer.stop();
+
+  EXPECT_GE(count, 3);
```

Comment:
Are you sure that stop and wakeup calls will not create jitter in the execution time of the two threads? If so, you might want to increase the PERIOD from 10 ms to e.g. 100 ms. Only increase PERIOD if really necessary, since this will increase the overall unit test time.

Reply:
Good point, this is why I am using `  EXPECT_GE` instead of exact values, as it is not deterministic.